parsec/runtime.h

```diff
@@ -322,6 +322,41 @@ int parsec_context_test( parsec_context_t* context );
  */
 int parsec_context_wait(parsec_context_t* context);
 
+/**
+ * @brief Complete the execution of a given PaRSEC taskpool
+ *
+ * @details
+ * Progress the execution context until the given taskpool reaches termination.
+ * Upon return from this function, all resources (threads and accelerators)
+ * associated with the corresponding context are left in a mode where they are
+ * active iff more taskpools are still active. The taskpool must be ready and
+ * registered with a started context.
+ *
+ * @param[inout] tp the taskpool to complete.
+ *
+ * @return * A negative number to signal an error. Any other value, aka. a positive
+ *           number (including 0), to signal successful completion of all work
+ *           associated with the taskpool.
+ */
+int parsec_taskpool_wait(parsec_taskpool_t* tp);
+
+/**
+ * @brief Allow the main thread to temporarily join the computation
+ *    by executing one (or less) task
+ *
+ * @details
+ * Try to progress the execution context until the given taskpool reaches termination.
+ * Upon return from this function, all resources (threads and accelerators)
+ * associated with the corresponding context are left in a mode where they are
+ * active. The taskpool must be ready and registered with a started context.
+ *
```

Comment:
Cannot the test set the task pool as completed (i.e., have the effect of an implicit wait), from that documentation I infer that the only correct code is to follow `taskpool_test` with `taskpool_wait` even when `taskpool_test` returned 0.

Reply:
is the following mockup correct and does that description prevents it? 

```
taskpool_test(tp)
context_wait(ctx)
```

Reply:
This is the expected usage, as `taskpool_test` does not alter the state of the context (aka. it remains active).